packages/calcite-components/src/components/list/list.tsx

```diff
@@ -269,11 +269,27 @@ export class List
   }
 
   @Listen("calciteListItemSelect")
-  handleCalciteListItemSelect(): void {
+  handleCalciteListItemSelect({ target, detail }: CustomEvent<{ multiple: boolean }>): void {
```

Comment:
Is `multiple` strictly for internal purposes? If so, can we prevent `multiple` from being exposed in the public event type?

Reply:
@jcfranco how would we do that? It might still be useful for public but correct its basically just for us.

Reply:
value-list did expose this as well as `shiftPressed`. https://github.com/Esri/calcite-design-system/blob/2b2506d07978b2fb66a63c1fed57c3f36c0d2996/packages/calcite-components/src/components/value-list-item/value-list-item.tsx#L192-L196

Reply:
I'm ok with it being public but i'll defer to @jcfranco. Not sure how to mark an event detail as `@internal`

Reply:
I can create and emit an internal event instead but then I'm not sure if its guaranteed the internal one would be handled first before updating the selected items.

Reply:
I think we could hide it from the public types by doing the following:

1. Type the item's select event emitter as `void`.
2. Use an internal type for the detail in the event handler (looks like `ListItemSelectDetail` in this case).
3. Use the internal type in the list's select event handler (as it currently does)
4. Not sure, but exporting the new event might no longer be needed as it wouldn't be used in the event type.

Based on the latest changes in this PR, you would need to do something like the following:

#### `list-item.tsx`

```ts
@Event({ cancelable: false }) calciteListItemSelect: EventEmitter<void>;
// ...
const internalDetail: ListItemSelectDetail = { 
  selectMultiple: shiftKey && selectionMode === "multiple" 
};
this.calciteListItemSelect.emit(internalDetail as any);
```

In addition, we could rename the prop in the interface to `_selectMultiple` to make it clear that it is internal.

WDYT?

Reply:
@jcfranco I updated to use a separate internal event. Can you review again?